core/src/common/components/ellipsis.tsx

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) 2021 Terminus, Inc.
+//
+// This program is free software: you can use, redistribute, and/or modify
+// it under the terms of the GNU Affero General Public License, version 3
+// or later ("AGPL"), as published by the Free Software Foundation.
+//
+// This program is distributed in the hope that it will be useful, but WITHOUT
+// ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+// FITNESS FOR A PARTICULAR PURPOSE.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+import React, { useState, useRef, useCallback } from 'react';
+import { Tooltip } from 'nusi';
+import { AbstractTooltipProps } from '../interface';
```

Comment:
Where is this interface file?

Reply:
It's the 'core/src/common/interface.d.ts'.

Reply:
It's the 'core/src/common/interface.d.ts'.